src/extensions/foot/footext_hooks.cpp

```diff
@@ -35,9 +35,94 @@
 #include "asserthandler.h"
 #include "debughandler.h"
 
+#include "infantry.h"
+#include "unit.h"
+
 #include "hooker.h"
 #include "hooker_macros.h"
+#include "tibsun_defines.h"
+
+/*
+* The Foot object initializes the passenger.
+* 
+* Injection objective function: FootClass::Unlimbo
+* 
+* @author: RossinCarlinx
+*/
+static void __stdcall FootClass_Initialize_Passengers(FootClass* this_ptr, TechnoTypeClassExtension* technotypeext) {
```

Comment:
1. This function can be moved to FootClassExtension as a member function.
2. The param of TechnoTypeClassExtension is not required as you can perform the extension type class fetch starting from `this_ptr->Techno_Type_Class()`

Reply:
This has been moved to the FootExt.cpp and the parameters have been removed, do you want to merge it?